### PR TITLE
Adding of app-gw ssl-policy and auth-cert + tests

### PR DESCRIPTION
**Pull request by EvgenyAgafonchikov**
- Added application-gateway ssl-policy and authentication-certificate commands
- Added related tests
- Added related recordings

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@EvgenyAgafonchikov__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by MikhailTryakhov** [comment]
@amarzavery could you please review this code block?

File: lib/commands/arm/network/network._js [hunk 1]
```diff
@@ -3252,6 +3252,149 @@ exports.init = function (cli) {
...
 
+  var appGatewaySslPolicy = appGateway.category('ssl-policy')
+    .description('Commands to manage application gateway ssl policy');
+
+  appGatewaySslPolicy.command('create [resource-group] [gateway-name]')
+    .description($('Create a application gateway ssl policy'))
+    .usage('[options] <resource-group> <gateway-name>')
+    .option('-g, --resource-group <resource-group>', $('the name of the resource group'))
+    .option('-w, --gateway-name <gateway-name>', $('the name of the application gateway'))
+    .option('-d, --disabled-ssl-protocols <disabled-ssl-protocols>', $('SSL protocols to be disabled on Application Gateway'))
```
Comment (amarzavery) [inline lib/commands/arm/network/network._js]: --disabled-ssl-protocols -> this should --disable~~d~~-ssl-protocols as they are not yet disabled. They will be once the command executes successfully. Can you also provide in the help what values can the user provide? From the name of the switch it looks like multiple values can be provided. If so then do you accept comma separated values? Please make sure all of the above questions are answered in the help.

File: lib/commands/arm/network/network._js [hunk 2]
```diff
@@ -3252,6 +3252,149 @@ exports.init = function (cli) {
...
+      resourceGroup = cli.interaction.promptIfNotGiven($('Resource group: '), resourceGroup, _);
+      gatewayName = cli.interaction.promptIfNotGiven($('Application gateway name: '), gatewayName, _);
+
+      var networkManagementClient = getNetworkManagementClient(options);
+      var appGateway = new AppGateway(cli, networkManagementClient);
+      appGateway.removeSslPolicy(resourceGroup, gatewayName, options, _);
+    });
+
+  appGatewaySslPolicy.command('show [resource-group] [gateway-name]')
+    .description($('Show a application gateway ssl policy'))
```
Comment (amarzavery) [inline lib/commands/arm/network/network._js]: Show **an** ...

File: lib/commands/arm/network/appGateway._js [hunk 3]
```diff
@@ -807,6 +808,190 @@ __.extend(AppGateways.prototype, {
...
+
+    if(!result.authenticationCertificates) {
+      result.authenticationCertificates = [];
+    }
+
+    var authCert = {};
+    authCert.name = name || 'default';
+
+    if(options.certFile) {
+      var data = fs.readFileSync(options.certFile);
```
Comment (amarzavery) [inline lib/commands/arm/network/appGateway._js]: all the sync versions throw an error. It is good to have a try catch around them and throw the original error back with a meaningful error message 'Error in reading the provided cert file. Make sure that the file path is correct and it has correct content.\n' + util.inspect(err, {depth: null});

File: lib/commands/arm/network/network._js (hunk 1 above)
Comment (EvgenyAgafonchikov) [inline lib/commands/arm/network/network._js]: Updated name and help message

File: lib/commands/arm/network/network._js (hunk 2 above)
Comment (EvgenyAgafonchikov) [inline lib/commands/arm/network/network._js]: Fixed

File: lib/commands/arm/network/appGateway._js (hunk 3 above)
Comment (EvgenyAgafonchikov) [inline lib/commands/arm/network/appGateway._js]: Updated

**Comment by MikhailTryakhov** [comment]
Thanks @amarzavery 
Hope that's fixed

**Pull request merged**